src/site/content/en/blog/payment-and-address-form-best-practices/index.md

```diff
@@ -0,0 +1,782 @@
+
+For validation, avoid using regular expressions that only match Latin characters. Latin-only excludes
+users with names or addresses that include characters that aren't in the Latin alphabet. Allow Unicode 
+letter matching instead—and ensure your back-end supports Unicode securely as both input and output. 
+Unicode in regular expressions is well supported by modern browsers.
+
+{% Compare 'worse' %}
+```html
+<!-- Names with non-Latin characters (such as Françoise or Jörg) are 'invalid'. -->
+<input pattern="[\w \-]+" ...>
+```
+{% endCompare %}
+
+{% Compare 'better' %}
+```html
+<!-- Accepts Unicode letters. -->
+<input pattern="[\p{L} \-]+" ...>
+```
+{% endCompare %}
+
+
+<figure class="w-figure">
+   <video controls autoplay loop muted class="w-screenshot">
+     <source src="https://samdutton.com/unicode-letter-matching.mp4" type="video/mp4">
+   </video>
+  <figcaption class="w-figcaption">Unicode letter matching compared to Latin-only letter matching.</figcaption>
+</figure>
+
+### Allow for a variety of address formats {: #address-variety}
+
+When you're designing an address form, bear in mind the bewildering variety of address formats, 
+even within a single country. Be careful not to make assumptions about "normal" addresses. (Take a 
+look at [UK Address Oddities!](http://www.paulplowman.com/stuff/uk-address-oddities/) if you're not 
+convinced!)
+
+#### Make address forms flexible {:flexible-address}
+
+Don't force users to [shoe-horn](https://en.wikipedia.org/wiki/Shoehorn#Turn_of_phrase) their address 
+into form fields that don't fit.
+
+For example, don't insist on a house number and street name in separate inputs, since many addresses 
+don't use that format, and incomplete data can break browser autofill.
+
+Be especially careful with `required` address fields. For example, addresses in large cities in the 
+UK do not have a county, but many sites still force users to enter one.
+
+Using two flexible address lines can work well enough for a variety of address formats.  
+
+```html
+<input autocomplete="address-line-1" id="address-line1" ...>
+<input autocomplete="address-line-2" id="address-line2" ...>
+```
+
+Add labels to match:
+
+```html/0-2,5-7
+<label for="address-line-1">
+Address line 1 (or company name)
+</label>
+<input autocomplete="address-line-1" id="address-line1" ...>
+
+<label for="address-line-2">
+Address line 2 (optional)
+</label>
+<input autocomplete="address-line-2" id="address-line2" ...>
+```
+You can try this out by remixing and editing the demo embedded below.
+
+{% Aside 'caution' %}
+Research shows that [**Address line 2** may be problematic for users]((https://baymard.com/blog/address-line-2)). 
+Bear this in mind when designing address forms—you should consider alternatives such as using a 
+single `textarea` (see below) or other options.
+{% endAside %}
+
+
+#### Consider using a single textarea for address {: #address-textarea}
```

Comment:
Are you confident that `textarea` works well with `autocomplete`? I saw it autofill works with `street-address`, but what if the user  has nothing to autofill? The user will probably enter the text freely and the address data will not be reusable at all. Also I've never seen address field as a `textarea` before. (Not sure about outside of Japan though)

Reply:
I warn about the potential autofill problem in the article (`street-address` versus `address-line1`, etc.) although Chrome autofills a `textarea` with `autocomplete="street-address"` correctly. You can try this out at [address-form.glitch.me](https://address-form.glitch.me). In Chrome settings, you manage your street address as a single field, and AFAICT that's what's stored from forms with `address-line1`, etc. I'll check all this with Ali.

Whatever, I take your point — it is quite unusual to see `textarea` used like this. I think it's a good idea (there are a few UI/UX posts online about this) but if it's not what people expect, maybe we should be more careful about recommending it. 

I've used `textarea` in the codelab, but with a warning that in the real world you'll need to localize address forms. It's hard to provide an example address form that makes sense for a global audience.

There's also [research from Baymard](https://baymard.com/blog/address-line-2) (mentioned in the codelab) that shows **Address line 2** confuses users.